quilt/tools/build.py

```diff
@@ -15,12 +16,39 @@
 
 from .const import DEFAULT_BUILDFILE, PACKAGE_DIR_NAME, PARSERS, RESERVED
 from .core import PackageFormat, BuildException, exec_yaml_python, load_yaml
+from .hashing import digest_file, digest_string
 from .package import Package, ParquetLib
 from .store import PackageStore, VALID_NAME_RE, StoreException
 from .util import FileWithReadProgress
 
 from . import check_functions as qc            # pylint:disable=W0611
 
+def _have_pyspark():
+    """
+    Check if we're running Pyspark
+    """
+    if _have_pyspark.flag is None:
+        try:
+            if Package.get_parquet_lib() is ParquetLib.SPARK:
+                import pyspark  # pylint:disable=W0612
+                _have_pyspark.flag = True
+            else:
+                _have_pyspark.flag = False
+        except ImportError:
+            _have_pyspark.flag = False
+    return _have_pyspark.flag
+_have_pyspark.flag = None
+
+def _path_hash(path, transform, kwargs):
+    """
+    Generate a hash of source file path + transform + args
+    """
+    sortedargs = ["{k}:{v}".format(k=key, v=value) for key, value in sorted(iteritems(kwargs))]
```

Comment:
This will hash different types as the same thing - e.g., `dict(a=1)` and `dict(a='1')`.

You could do something like `["%r:%r" % item for item in sorted(iteritems(kwargs))]`. Though even that looks fragile to me... Don't really have better ideas.